### ISSUE TITLE
Squiz.Commenting.FunctionComment.InvalidNoReturn with never return type

### ISSUE THREAD

**Issue by SharkMachine** [squizlabs/PHP_CodeSniffer#3798]
**Describe the bug**

`Squiz.Commenting.FunctionComment` doesn't seem to support PHP 8.1 `never` return type.

**Code sample**
```php
<?php

/**
 * @return never
 */
function redirect(): never
{
    header('Location: /');
    exit();
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <description>Coding standard.</description>
    <arg name="tab-width" value="4"/>

    <rule ref="PSR12" />

    <!-- PHPDoc and type hinting sniffs -->
    <rule ref="Squiz.Commenting.FunctionComment" />
    <exclude-pattern>vendor/*</exclude-pattern>
</ruleset>
```

**To reproduce**

1. Create a file called `test.php` with the code sample above.
2. Create a file called `phpcs.xml` with the custom ruleset above.
3. Run `phpcs test.php`
4. See error message displayed
```
FILE: test.php
------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------
 4 | ERROR | Function return type is not void, but function has no return statement
------------------------------------------------------------------------------------

Time: 34ms; Memory: 6MB
```

**Expected behavior**

`never` return type is detected and no error is shown.

**Versions (please complete the following information):**

 - OS: Ubuntu 22.04 LTS
 - PHP: 8.1
 - PHPCS: 3.7.2
 - Standard: Squiz

**Comment by fredden** [comment]
https://github.com/squizlabs/PHP_CodeSniffer/pull/3717 looks like it will fix this.

**Comment by jrfnl** [comment]
@SharkMachine Could you please verify ? The fix from #3717 was included in the [3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0) from a few weeks ago.

**Comment by SharkMachine** [comment]
The fix works
